.github/workflows/shippable_builds.yml

```diff
@@ -224,21 +225,56 @@ jobs:
 
           ./gradlew :app-${APP_NAME}:printVersionInfo -PbuildType=${RELEASE_TYPE} -PflavorName=${PACKAGE_FLAVOR} --configure-on-demand
 
-      - name: Bump version code
-        id: bump_version_code
-        if: ${{ contains(matrix.releaseTarget, 'github') }}
+      - name: Determine new version code
+        id: new_version_code
+        if: ${{ contains(matrix.releaseTarget, 'github') || needs.dump_config.outputs.releaseType == 'daily' }}
         shell: bash
         env:
           APP_NAME: ${{ matrix.appName }}
           OLD_VERSION_CODE: ${{ steps.appinfo.outputs.VERSION_CODE }}
+          RELEASE_TYPE: ${{ vars.RELEASE_TYPE }}
         run: |
-          NEW_VERSION_CODE=$(($OLD_VERSION_CODE + 1))
+          date_version_code() {
+            # Daily uses date-based calculation for version code in yDDDHHmm format
+            # y: Years since 2023 (e.g. 2024 is 1)
+            # DDD: Day of the year (3 digits, zero-padded)
+            # HH: Hour of the day (2 digits, 00-23)
+            # mm: Minute of the hour (2 digits)
+            datetime="$1"
+            year=$(( $(date -d "${datetime}" +"%Y") - 2023 ))
+            day=$(printf "%03d" $(date +"%j"))
+            time=$(date +"%H%M")
+            echo "${year}${day}${time}"
+          }
+          if [[ "$RELEASE_TYPE" = "beta" || "$RELEASE_TYPE" = "release" ]]; then
+            NEW_VERSION_CODE=$(($OLD_VERSION_CODE + 1))
+          elif [[ "$RELEASE_TYPE" = "daily" ]]; then
+            VERSION_FILE="$(pwd)/version.properties"
+            if [[ -e "$VERSION_FILE" ]]; then
+              while IFS="=" read -r key value; do
+                if [[ "${key}" = "new_version_code" ]]; then
+                  NEW_VERSION_CODE="${value}"
+                fi
+              done < $VERSION_FILE
+            else
+              NEW_VERSION_CODE=$(date_version_code "now")
+              echo "new_version_code=${NEW_VERSION_CODE}" > $VERSION_FILE
+            fi
+          fi
+          # Ensure version code is not higher than max version code
+          MAX_VERSION_DATE="2025-12-31 23:59"
+          MAX_VERSION_CODE=$(date_version_code "$MAX_VERSION_DATE")
```

Comment:
Re version.properties: Ah, yeah of course..  maybe an upload artifact would be the right approach. I'll try that out.

Re version code: Well, hopefully that code isn't wrong! But yes, I can hardcode it instead.

Reply:
Re version.properties - If you calculate the date in an early job that is not parallel, you should be able to use that going forward, dump_environment would work for that.

Reply:
And if you could add a check for release/beta that the version code is not higher than 30, we should be good until end of 2025.

Reply:
I'm testing the new commits I've pushed. I'll let you know when they're ready for review.

Reply:
This is ready for review again. I'm using upload-artifact and download-artifact to store version.properties for parallel matrix jobs. It's not atomic but it should be good enough if there's a race. I had to use "continue-on-error: true" for both actions because they fail if the file exists (upload-artifact) or doesn't exist (download-artifact). I don't love it but it seems to work.

Reply:
Another option that would work better is to introduce another job before release_commit and let release_commit use the output from it. I just don't know if we want another job.

Reply:
Sorry, I just saw your latest comments. I'll rework with your comments.